Catalog/Examples/Networking/Japx/NetworkingJapxPresenter.swift

```diff
@@ -0,0 +1,145 @@
+//
+//  NetworkingJapxPresenter.swift
+//  Catalog
+//
+//  Created by Mate Masnov on 02/12/2019.
+//  Copyright (c) 2019 Infinum. All rights reserved.
+//
+//  This file was generated by the 🐍 VIPER generator
+//
+
+import UIKit
+
+final class NetworkingJapxPresenter {
+
+    // MARK: - Private properties -
+
+    private unowned let _view: NetworkingJapxViewInterface
+    private let _wireframe: NetworkingJapxWireframeInterface
+    private let _interactor: NetworkingJapxInteractorInterface
+
+    private var _user: PokedexUser?
+    private var _authToken: String?
+    
+    // MARK: - Lifecycle -
+
+    init(wireframe: NetworkingJapxWireframeInterface, view: NetworkingJapxViewInterface, interactor: NetworkingJapxInteractorInterface) {
+        _wireframe = wireframe
+        _view = view
+        _interactor = interactor
+    }
+}
+
+// MARK: - Extensions -
+
+extension NetworkingJapxPresenter: NetworkingJapxPresenterInterface {
+    
+    func didPressCreate(with email: String?, username: String?, password: String?) {
+        guard let email = email, let username = username, let password = password else {
+            _wireframe.showFailure(with: "Error", message: "Invalid credentials!")
+            return
+        }
+        
+        _view.showLoading()
+        
+        _interactor
+            .createUser(
+                email: email,
+                username: username,
+                password: password,
+                passwordConfirmation: password
+            ) { [unowned _wireframe, unowned _interactor, unowned _view, weak self] (result) in
```

Comment:
Not sure why you are using unowned here, but weak should be used instead. This is not Rx so you don't control lifecycle of API call thus issue may occur if you go back from screen before API call finished, and when it finally completes your wireframe, interactor and view will be gone thus causing the crash